src/tokens.rs

```diff
@@ -275,76 +291,114 @@ where
 
     fn new_transform(&self, service: S) -> Self::Future {
         ok(TokenParserMiddleware {
-            service,
+            service: Rc::new(Mutex::new(service)),
             inner: self.0.clone(),
         })
     }
 }
 
 /// TokenParser middleware
 pub struct TokenParserMiddleware<S> {
-    service: S,
+    service: Rc<Mutex<S>>,
```

Comment:
Any reason to use `Mutex` instead of `RefCell`/`Cell`/`OnceCell`?

Reply:
RefCell should work. Fixed.